### PR TITLE
add unavailable dates feature

### PR DESCRIPTION
**Pull request by rarescrisan**
Add `unavailableDates` prop to context that allows a consumer to supply an array of dates that can be marked as unavailable.

work was mostly introduced here:
packages/styled/src/components/Day/Day.tsx
packages/hooks/src/useDay/useDay.ts
packages/styled/src/context/datepickerContext.ts

### CODE REVIEW

File: packages/styled/src/components/Datepicker/Datepicker.tsx [hunk 1]
```diff
@@ -266,6 +268,7 @@ function Datepicker(
     scrollTopToMonthGrid()
   }
 
+  console.log('BLOCKED ARRAY', unavailableDates)
```
Comment (tresko) [inline packages/styled/src/components/Datepicker/Datepicker.tsx]: Remove console.log

File: packages/styled/src/components/Day/Day.tsx [hunk 2]
```diff
@@ -321,6 +325,8 @@ function Day({day, date}: DayProps) {
     [dayProps.isSelected, dayProps.isSelectedStartOrEnd, theme, dayProps.isWithinHoverRange],
   )
 
+  console.log('DAY', day)
```
Comment (tresko) [inline packages/styled/src/components/Day/Day.tsx]: Remove console.log

File: packages/styled/src/components/Datepicker/Datepicker.stories.tsx [hunk 3]
```diff
@@ -76,6 +78,7 @@ function App({
     }
   }
 
+  console.log('APP BLOCK', unavailableDates)
```
Comment (tresko) [inline packages/styled/src/components/Datepicker/Datepicker.stories.tsx]: Remove console.log

File: packages/hooks/src/useDay/useDay.ts [hunk 4]
```diff
@@ -27,16 +29,29 @@ function useDay({
...
   onDateHover,
   dayRef,
+  unavailableDates,
 }: UseDayProps) {
   const onClick = useCallback(() => onDateSelect(date), [date, onDateSelect])
   const onMouseEnter = useCallback(() => onDateHover(date), [date, onDateHover])
+
+  function isInUnavailableDates(unavailableDates: Date[], date: Date) {
+    let isBlocked = false
+    unavailableDates.forEach(_date => {
```
Comment (tresko) [inline packages/hooks/src/useDay/useDay.ts]: Use reduce function instead forEach

File: packages/hooks/src/useDay/useDay.ts [hunk 5]
```diff
@@ -13,6 +14,7 @@ interface UseDayProps {
   onDateSelect(date: Date): void
   onDateHover(date: Date): void
   dayRef: React.RefObject<HTMLButtonElement>
+  unavailableDates: Date[]
```
Comment (tresko) [inline packages/hooks/src/useDay/useDay.ts]: This prop should be optional.

File: packages/styled/src/components/Datepicker/Datepicker.stories.tsx (hunk 3 above)
Comment (rarescrisan) [reply packages/styled/src/components/Datepicker/Datepicker.stories.tsx]: sry, done

File: packages/hooks/src/useDay/useDay.ts (hunk 4 above)
Comment (rarescrisan) [reply packages/hooks/src/useDay/useDay.ts]: used a .some

File: packages/hooks/src/useDay/useDay.ts (hunk 5 above)
Comment (rarescrisan) [reply packages/hooks/src/useDay/useDay.ts]: done

File: packages/styled/src/components/Day/Day.tsx (hunk 2 above)
Comment (rarescrisan) [reply packages/styled/src/components/Day/Day.tsx]: sry, done

File: packages/styled/src/components/Datepicker/Datepicker.tsx (hunk 1 above)
Comment (rarescrisan) [reply packages/styled/src/components/Datepicker/Datepicker.tsx]: sry, done

**Pull request merged**